src/psyclone/domain/gocean/nodes/gocean_extract_node.py

```diff
@@ -209,24 +255,48 @@ def create_driver(self, input_list, output_list):
                 # comparison with the post value works as expected
                 # TODO #644 - create the right "0.0" type here (e.g.
                 # 0.0d0, ...)
-                assign = AssignGen(prog, var_name, "0.0d0")
+                assign = AssignGen(prog, local_name, "0.0d0")
                 prog.add(assign)
 
         # Now add the region that was extracted here:
         prog.add(CommentGen(prog, ""))
         prog.add(CommentGen(prog, " RegionStart"))
 
-        # For the driver we have to re-create the code, but
-        # the arguments are not fields anymore, but simple arrays.
-        # So we need to make sure that the field parameters
-        # are changed from "fld%data" to just "fld". This is
-        # achieved by temporary changing the value of the
-        # "go_grid_data" property (e.g. "{0}%data") to just "{0}".
+        # For the driver we have to re-create the code of the
+        # instrumented region, but in this stand-alone driver the
+        # arguments are not dl_esm_inf fields anymore, but simple arrays.
+        # Similarly, for properties we cannot use e.g. 'fld%grid%dx'
+        # anymore, we have to use e.g. a local variable 'dx' that has
+        # been created. Since we are using the existing way of creating
+        # the code for the instrumented region, we need to modify how
+        # these variables are created. We do this by temporarily
+        # modifying the properties in the config file.
         api_config = Config.get().api_conf("gocean1.0")
-        props = api_config.grid_properties
-        old_data_property = props["go_grid_data"]
-        Property = namedtuple("Property", "fortran type")
-        props["go_grid_data"] = Property("{0}", "array")
+        all_props = api_config.grid_properties
+        # Keep a copy of the original values, so we can restore
+        # them later
+        orig_props = dict(all_props)
+
+        # 1) A grid property is defined like "{0}%grid%dx". This is
+        #    changed to be just 'dx', i.e. the final component of
+        #    the current value (but we also take renaming into account,
+        #    so 'dx' might become 'dx_0').
+        #    If a property is not used, it doesn't matter if we modify
+        #    its definition, so we just change all properties.
+        for name, prop in all_props.items():
+            last_percent = prop.fortran.rfind("%")
+            if last_percent > -1:
+                # Get the last field name, which will be the
+                # local variable name
+                local_name = prop.fortran[last_percent+1:]
+                unique_name = rename_variable.get(local_name, local_name)
+                all_props[name] = api_config.make_property(unique_name,
+                                                           prop.type)
+
+        # 2) grid_data is "{0}%data" --> this just becomes {0}
```

Comment:
s/grid_data/field_data/ I think? After much head scratching, is this the thing we had a lot of debate over a few PRs ago? If so, I accept it is called `grid_data` but please make the comment clear that this actually refers to data on the grid, i.e. fields.

Reply:
I think we agreed for now to keep calling it grid_property, till we have time to split this into grid- and field-properties. But I've added comments to clarify this here.